velox/functions/prestosql/types/TimestampWithTimeZoneType.cpp

```diff
@@ -53,15 +57,45 @@ void castFromTimestamp(
   });
 }
 
+void castFromString(
```

Comment:
I think this should work pretty much out of the box if you just provide a few more Converter<> specializations, so you may not need any of this code. Have you tried that? 

Check velox/type/Conversions.h

Reply:
I don't see any implementation for `TimestampWithTimezone` conversion there, do you want me to move the conversion (lines 74-77) there?

Reply:
Not there, but why can't you provide them here (in this compilation unit)?

Reply:
@pedroerp Added new Converter specialization here, please let me know if it looks good. If so, I can do the same for the existing `castFromTimestamp()` implementation.